examples/firmware_update.py

```diff
@@ -0,0 +1,111 @@
+#!/bin/python3
+
+import dronecan, time, math
+from argparse import ArgumentParser
+
+import sys
+import os
+
+# import logging
+# logging.basicConfig(level=logging.DEBUG)
+
+parser = ArgumentParser(description='dump all DroneCAN messages')
+parser.add_argument("--bitrate", default=1000000, type=int, help="CAN bit rate")
+parser.add_argument("--node-id", default=100, type=int, help="CAN node ID")
+parser.add_argument("--dna-server", action='store_true', default=True, help="run DNA server")
+parser.add_argument("--port", default='/dev/ttyACM0', type=str, help="serial port")
+parser.add_argument("--app-firmware", default=None, type=str, help="serial port")
+
+args = parser.parse_args()
+
+firmware_path = args.app_firmware
```

Comment:
The symlinking doesn't work unless it's an absolute path.

```suggestion
firmware_path = os.path.abspath(args.app_firmware)
```